tasks/species_typing/task_tbprofiler_output_parsing.wdl

```diff
@@ -104,166 +219,341 @@ task tbprofiler_output_parsing {
+          return "Uncertain significance" if interpretation_destination == "LIMS" else "U"
+        else:
+            if substitution_type != "synonymous_variant" or "upstream_gene_variant" in substitution_type:
+              return "S" if interpretation_destination == "MDL" else "U"
+            else:
+              return "S"
+
+      elif gene == "rpoB": # apply expert rules 2.2.2
+        if SPECIAL_POSITIONS[gene][1] <= position_nt <= SPECIAL_POSITIONS[gene][2]:
+            if substitution_type != "synonymous_variant":
+              return "Assoc with R" if interpretation_destination == "LIMS" else "R"
+            else:
+              return "S"   
+        else:
+            if substitution_type != "synonymous_variant" or "upstream_gene_variant" in substitution_type:
+              return "S" if interpretation_destination == "MDL" else "U"
+            else:
+              return "S"
+
+      elif gene not in GENE_LIST_OPTION_1 or gene not in GENE_LIST_OPTION_2: # NOT AN EXPERT RULE: 3.2
+        if substitution_type != "synonymous_variant" or "upstream_gene_variant" in substitution_type:
+          return "Snoexpert" if interpretation_destination == "MDL" else "Unoexpert"
+        else:
+          return "Snoexpert"
+
+      return ""
+
+    def remove_no_expert(row):
       """
-      This function takes the annotation of resistance by TBProfiler and 
-      returns simple R (resistant), U (uncertain) and S (susceptible) notations
+      This function removes the 'noexpert' suffix in the case where 
+      the logic applied is not considered an expert rule.
       """
-      if annotation == "Not assoc w R":
-        return "S"
-      elif annotation == "Uncertain significance":
-        return "S"
-      elif annotation == "Assoc w R - interim":
-        return "U"
-      elif annotation == "Assoc w R":
-        return "R"
-      else:
-        return "S"
+      if "noexpert" in row["looker_interpretation"]:
+        interpretation = row["looker_interpretation"]
+        row["looker_interpretation"] = interpretation.replace("noexpert", "")
+        row["rationale"] = "No WHO annotation or expert rule"
+      
+      if "noexpert" in row["mdl_interpretation"]:
+        interpretation = row["mdl_interpretation"]
+        row["mdl_interpretation"] = interpretation.replace("noexpert", "")
+        row["rationale"] = "No WHO annotation or expert rule"
+      
+      return row
 
-    def get_lineage(json_file):
+    def get_lineage_LIMS(json_file):
+      """
+      This function returns the lineage in English for the LIMS report
+      """
       with open(json_file) as js_fh:
         results_json = json.load(js_fh)
         if results_json["main_lin"] == "":
-          return "DNA of M. tuberculosis complex not detected"
-        elif results_json["main_lin"] == "M.bovis":
+          return "No main lineage could be identified"
```

Comment:
Switch "DNA from M.Tb found" is it passes the coverage report, otherwise, write "DNA of M. tb complex not detected"

Reply:
Please confirm @frankambrosio3